src/braket/aws/aws_device.py

```diff
@@ -320,33 +327,31 @@ def _aws_session_for_qpu(device_arn: str, aws_session: Optional[AwsSession]) ->
         See `braket.aws.aws_qpu.AwsDevice.DEVICE_REGIONS` for the
         AWS Regions the devices are located in.
         """
-        region_key = device_arn.split("/")[-2]
-        qpu_regions = AwsDevice.DEVICE_REGIONS.get(region_key, [])
-        return AwsDevice._copy_aws_session(aws_session, qpu_regions)
+        return AwsDevice._copy_aws_session(aws_session, AwsDevice.QPU_REGIONS.get(device_arn), None)
 
     @staticmethod
     def _copy_aws_session(
         aws_session: Optional[AwsSession],
-        regions: List[str] = None,
+        region: Optional[str] = None,
         max_connections: Optional[int] = None,
     ) -> AwsSession:
         config = Config(max_pool_connections=max_connections) if max_connections else None
         if aws_session:
             session_region = aws_session.boto_session.region_name
-            new_regions = regions or [session_region]
-            if session_region in new_regions and not config:
+            new_region = region or session_region
+            if session_region == new_region and not config:
```

Comment:
I'm actually wondering if this is desired behavior here - do we want to enable refreshing credentials for a new AwsSession? (Probably not in this PR, but worth thinking about later.)

Reply:
That's something for a future PR; all this change does is move from lists of regions to just individual regions.